### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by joeuser846** [lomori/spherorvr-ros2#1]
I get this colcon build error on my fork of your repo:

/usr/bin/python3: can't open file '/opt/ros/iron/ros2_ws/spherorvr-ros2/sphero/setup.py': [Errno 2] No such file or directory

I'm new to ROS2 so maybe I'm missing something?  Thanks for any help.

**Comment by lomori** [comment]
Yes, you are right, it's not building. It has been a while that I worked on that one and I myself was new to ROS2. In fact, if I recall correctly, I've used that project to learn ROS2. I will need to refactor it. Meanwhile if you want to proceed, it might be easier to just create an empty ROS2 project and add the code there. Sorry about that!
Regards,Luiz
    On Sunday, October 8, 2023 at 11:41:20 AM EDT, Joe User ***@***.***> wrote:  
 
 


I get this colcon build error on my fork of your repo:

/usr/bin/python3: can't open file '/opt/ros/iron/ros2_ws/spherorvr-ros2/sphero/setup.py': [Errno 2] No such file or directory

I'm new to ROS2 so maybe I'm missing something? Thanks for any help.

**Comment by joeuser846** [comment]
Thanks for the suggestion.  I was able to create a new ROS2 package which
built the required template files.  Then I bumped into the next issue: how
to include the Sphero SDK that your code references.  I installed it with
pip but that doesn’t seem to make it visible to your Python code.  I’m now
researching how to merge standard Python code into a ROS2 package.

I’ve reviewed all the available ROS1 and ROS2 packages for Sphero (RVR+)
and yours looks the most complete because it supports both odom publish and
cmd_vel subscription thus enabling real  navigation.  I’d love to get it
working!

Thanks again.

**Comment by lomori** [comment]
Thanks for your comments and again, and so sorry for the problems.
The cool thing I found about Sphero is that they did a good job in their base movement control. Not sure if you saw it, but I did a video with Sphero, Localization, and Navigation using a (more or less) cheap 2D Lidar. After mapping an environment, it's able to navigate from any A to any B avoiding walls, chairs, etc. I haven't tested it, but it's supposed to avoid blocks during run-time, like a person moving around.
    On Tuesday, October 10, 2023 at 12:29:24 PM EDT, Joe User ***@***.***> wrote:  
 
 

Thanks for the suggestion. I was able to create a new ROS2 package which
built the required template files. Then I bumped into the next issue: how
to include the Sphero SDK that your code references. I installed it with
pip but that doesn’t seem to make it visible to your Python code. I’m now
researching how to merge standard Python code into a ROS2 package.

I’ve reviewed all the available ROS1 and ROS2 packages for Sphero (RVR+)
and yours looks the most complete because it supports both odom publish and
cmd_vel subscription thus enabling real navigation. I’d love to get it
working!

Thanks again.
